Gems/Archive/Code/Source/Clients/ArchiveReader.cpp

```diff
@@ -0,0 +1,928 @@
+            static_cast<AZ::u64>(archiveFileToken)];
+
+        if (filePathOffsetSize.m_size == 0)
+        {
+            ArchiveListFileResult errorResult;
+            errorResult.m_filePathToken = archiveFileToken;
+            errorResult.m_resultOutcome = AZStd::unexpected(
+                ResultString::format(R"(A file token "%llu" is being used to extract the file,)"
+                    " but the file path stored in the TOC is empty."
+                    "This indicates that the token is referring to a deleted file",
+                    static_cast<AZ::u64>(archiveFileToken)));
+            return errorResult;
+        }
+
+        // The file has been found and has a non-empty path
+        // Populate the ArchiveListFileResult structure
+        ArchiveListFileResult listResult;
+        // Extract the path stored in the file path blob into the extract result
+        listResult.m_relativeFilePath = AZ::IO::PathView(
+            m_archiveToc.m_tocView.m_filePathBlob.substr(filePathOffsetSize.m_offset, filePathOffsetSize.m_size));
+        listResult.m_filePathToken = archiveFileToken;
+
+        // Gather the file metadata
+        const ArchiveTocFileMetadata& fileMetadata = m_archiveToc.m_tocView.m_fileMetadataTable[
+            static_cast<AZ::u64>(archiveFileToken)];
+
+        // Use the compression algorithm index to lookup the compression algorithm ID
+        // if the file the value is less than the size of the compression AlgorithmIds array
+        if (fileMetadata.m_compressionAlgoIndex < m_archiveHeader.m_compressionAlgorithmsIds.size())
+        {
+            listResult.m_compressionAlgorithm = m_archiveHeader.m_compressionAlgorithmsIds[
+                fileMetadata.m_compressionAlgoIndex];
+        }
+
+        listResult.m_uncompressedSize = fileMetadata.m_uncompressedSize;
+        if (auto rawFileSizeResult = GetRawFileSize(fileMetadata, m_archiveToc.m_tocView.m_blockOffsetTable);
+            !rawFileSizeResult)
+        {
+            ArchiveListFileResult errorResult;
+            errorResult.m_filePathToken = archiveFileToken;
+            // Take the error from GetRawFileSize call and return that
+            errorResult.m_resultOutcome = AZStd::unexpected(AZStd::move(rawFileSizeResult.error()));
+            return errorResult;
+        }
+        else
+        {
+            listResult.m_compressedSize = rawFileSizeResult.value();
+        }
+        listResult.m_offset = fileMetadata.m_offset;
+        listResult.m_crc32 = fileMetadata.m_crc32;
+
+        return listResult;
+    }
+
+    ArchiveListFileResult ArchiveReader::ListFileInArchive(AZ::IO::PathView relativePath) const
+    {
+        if (relativePath.empty())
+        {
+            ArchiveListFileResult errorResult;
+            errorResult.m_resultOutcome = AZStd::unexpected(
+                ResultString(R"(An empty file path has been supplied."
+                " An empty path cannot be found in the archive.)"));
```

Comment:
will the `R"` string literal span across two string literals that are broken across lines? Probably not needed at all here

Reply:
It will expand across multiple lines, but the raw string literal isn't needed here, so I'll remove it.